src/InMemoryRepository.php

```diff
@@ -256,11 +269,55 @@ private function removeResource(PuliResource $resource)
 
         unset($this->resources[$path]);
 
+        if ($resource instanceof LinkResource && array_key_exists($path, $this->links)) {
```

Comment:
You can change this to `isset($this->links[$path])` which performs slightly better.